src/ray/core_worker/core_worker.cc

```diff
@@ -873,7 +874,7 @@ Status CoreWorker::BuildArgsForExecutor(const TaskSpecification &task,
         metadata = std::make_shared<LocalMemoryBuffer>(
             const_cast<uint8_t *>(task.ArgMetadata(i)), task.ArgMetadataSize(i));
       }
-      args->at(i) = std::make_shared<RayObject>(data, metadata);
+      args->at(i) = std::make_shared<RayObject>(data, metadata, /*copy_data*/ true);
```

Comment:
FYI @edoakes @stephanie-wang this fixes https://github.com/ray-project/ray/issues/6264